src/index.ts

```diff
@@ -305,26 +311,42 @@ export const createClaimWithChildInstance = (
       );
       return tx;
     },
-    getClaimingIDsForUser: async (walletID: string): Promise<string[]> => {
+    getClaimingIDsForUser: async (
+      walletID: string
+    ): Promise<IDWithClaimStatus[]> => {
       const claimingToken = await getClaimingToken(
         config.web3Provider,
         config.claimingRegistrarAddress
       );
-      let childDomain: ethers.BigNumber;
-      const claimingIDs: string[] = [];
+      const IDs: IDWithClaimStatus[] = [];
       const bigNumDomainsOwned = await claimingToken.balanceOf(walletID);
       const numDomainsOwned = bigNumDomainsOwned.toNumber();
+      const promises: Promise<void>[] = [];
       if (numDomainsOwned != 0) {
-        let i = 0;
-        while (i < numDomainsOwned) {
-          childDomain = await claimingToken.tokenOfOwnerByIndex(walletID, i);
-          if (await instance.canBeClaimed(childDomain.toHexString())) {
-            claimingIDs.push(childDomain.toHexString());
-          }
-          i++;
+        let count = 0;
+        while (count < numDomainsOwned) {
+          const getDomain = async () => {
+            const childDomain = await claimingToken.tokenOfOwnerByIndex(
+              walletID,
+              count
+            );
+            const idWithClaimStatus: IDWithClaimStatus = {
+              id: childDomain.toHexString(),
+              canBeClaimed: await instance.canBeClaimed(
+                childDomain.toHexString()
+              ),
+            };
+            IDs.push(idWithClaimStatus);
+          };
+          promises.push(getDomain());
+          count++;
         }
       }
-      return claimingIDs;
+      await chunkedPromiseAll(promises, 100, 5); // chunks of 100 at a time, 5 ms delay between chunks
+      return [
```

Comment:
Isn't this the same thing as `[...IDs]` or just returning `IDs`

Reply:
i wanted the claimable IDs to always be returned first in the list - I suppose that's not strictly necessary and we could leave it up to the dapp to sort if they chose to